Sources/Datadog/DatadogInternal/DatadogCoreProtocol.swift

```diff
@@ -46,6 +46,22 @@ public protocol DatadogCoreProtocol {
 }
 
 extension DatadogCoreProtocol {
+    /// Registers a feature instance by its type description.
+    ///
+    /// - Parameter instance: The feaure instance to register
+    func register<T>(feature instance: T?) {
+        registerFeature(named: String(describing: T.self), instance: instance)
+    }
```

Comment:
Features' name in v1 does not make much sense, I'm thinking of using this method for all v1 features!

Reply:
Agree 👍, although shouldn't it replace:
```
func registerFeature(named featureName: String, instance: Any?)
```
in `DatadogCoreProtocol`, instead of forwarding? Now we have some features registered with `register(feature:)` and some with `registerFeature(named:instance:)`. It is very dangerous as both are not compatible and we need to know their registration method to use matching variant when obtaining the instance. Even if this is temporary shim between V1 and V2, I'd keep it straight as it can get messy.